### ISSUE TITLE
EPUBLISHCONFLICT, Cannot publish over existing version.

### ISSUE THREAD

**Issue by devTristan** [npm/npm#7773]
Something isn't right here. Note the version shown in `npm show`, while `1.3.0` is visible under the `time` object.

```
~/smartystreets$ grep version package.json 
  "version": "1.3.0",
~/smartystreets$ npm --version
2.5.1
~/smartystreets$ npm publish
npm ERR! Linux 3.16.0-31-generic
npm ERR! argv "node" "/redacted/usr/bin/npm" "publish"
npm ERR! node v0.10.25
npm ERR! npm  v2.5.1
npm ERR! code EPUBLISHCONFLICT

npm ERR! publish fail Cannot publish over existing version.
npm ERR! publish fail Update the 'version' field in package.json and try again.
npm ERR! publish fail 
npm ERR! publish fail To automatically increment version numbers, see:
npm ERR! publish fail     npm help version

npm ERR! Please include the following file with any support request:
npm ERR!     /redacted/smartystreets/npm-debug.log
~/smartystreets$ npm show smartystreets
 
{ name: 'smartystreets',
  description: 'Fast SmartyStreets geocoder',
  'dist-tags': { latest: '1.2.0' },
  versions: 
   [ '0.2.0',
     '0.3.0',
     '0.4.0',
     '0.4.1',
     '0.4.2',
     '0.4.3',
     '0.4.4',
     '0.5.0',
     '0.5.1',
     '0.5.2',
     '0.5.3',
     '0.5.4',
     '0.6.0',
     '0.6.1',
     '0.6.3',
     '0.6.4',
     '0.6.6',
     '0.6.7',
     '0.6.8',
     '0.6.9',
     '0.6.10',
     '0.6.11',
     '0.6.12',
     '0.7.0',
     '0.7.1',
     '0.8.0',
     '1.0.0-rc.0',
     '1.0.0-rc.1',
     '1.0.0-rc.2',
     '1.0.0-rc.3',
     '1.0.0-rc.4',
     '1.0.0-rc.5',
     '1.0.0-rc.6',
     '1.0.0-rc.7',
     '1.0.0-rc.8',
     '1.0.0-rc.9',
     '1.0.0',
     '1.0.1',
     '1.1.0',
     '1.2.0' ],
  maintainers: 'tristandavies <npm@tristan.io>',
  time: 
   { modified: '2015-03-26T18:10:53.161Z',
     created: '2014-07-07T20:55:02.401Z',
     '0.2.0': '2014-07-07T20:55:02.401Z',
     '0.3.0': '2014-07-07T20:56:24.795Z',
     '0.4.0': '2014-07-09T17:06:33.167Z',
     '0.4.1': '2014-07-09T18:45:21.940Z',
     '0.4.2': '2014-07-09T19:38:20.238Z',
     '0.4.3': '2014-07-09T19:40:15.913Z',
     '0.4.4': '2014-07-13T20:58:51.997Z',
     '0.5.0': '2014-07-15T13:32:40.663Z',
     '0.5.1': '2014-07-15T18:48:20.584Z',
     '0.5.2': '2014-07-15T19:39:54.019Z',
     '0.5.3': '2014-07-15T20:18:42.169Z',
     '0.5.4': '2014-07-15T20:35:01.311Z',
     '0.6.0': '2014-07-17T15:47:30.675Z',
     '0.6.1': '2014-07-29T01:10:11.844Z',
     '0.6.3': '2014-07-29T01:52:02.883Z',
     '0.6.4': '2014-07-29T01:57:58.707Z',
     '0.6.6': '2014-07-29T02:23:37.360Z',
     '0.6.7': '2014-07-29T13:23:06.593Z',
     '0.6.8': '2014-07-29T16:20:30.557Z',
     '0.6.9': '2014-07-29T16:22:21.632Z',
     '0.6.10': '2014-08-04T21:33:19.218Z',
     '0.6.11': '2014-08-05T13:05:29.628Z',
     '0.6.12': '2014-08-07T15:32:54.221Z',
     '0.7.0': '2014-09-12T13:24:53.051Z',
     '0.7.1': '2014-12-18T19:12:02.801Z',
     '0.8.0': '2014-12-30T18:49:55.585Z',
     '1.0.0-rc.0': '2015-01-02T21:54:19.436Z',
     '1.0.0-rc.1': '2015-01-05T17:34:17.715Z',
     '1.0.0-rc.2': '2015-01-05T17:37:09.307Z',
     '1.0.0-rc.3': '2015-01-05T18:24:19.975Z',
     '1.0.0-rc.4': '2015-01-29T19:18:16.388Z',
     '1.0.0-rc.5': '2015-01-29T19:20:51.209Z',
     '1.0.0-rc.6': '2015-02-10T19:06:33.035Z',
     '1.0.0-rc.7': '2015-02-10T19:32:30.193Z',
     '1.0.0-rc.8': '2015-02-10T19:33:54.195Z',
     '1.0.0-rc.9': '2015-02-10T20:21:24.825Z',
     '1.0.0': '2015-02-27T21:13:18.475Z',
     '1.0.1': '2015-03-05T22:03:02.847Z',
     '1.1.0': '2015-03-20T15:24:26.786Z',
     '1.2.0': '2015-03-20T15:30:56.598Z',
     '1.3.0': '2015-03-26T18:03:19.649Z' },
  homepage: 'https://github.com/faradayio/node_smartystreets',
  repository: 
   { type: 'git',
     url: 'https://github.com/faradayio/node_smartystreets.git' },
  bugs: { url: 'https://github.com/faradayio/node_smartystreets/issues' },
  readmeFilename: 'README.md',
  version: '1.2.0',
  main: 'index.js',
  preferGlobal: 'true',
  bin: { smartystreets: 'program.js' },
  dependencies: 
   { async: '^0.9.0',
     commander: '^2.2.0',
     'fast-csv': '^0.5.6',
     request: '^2.36.0',
     terminus: '^1.0.11',
     through2: '^0.6.3',
     'through2-concurrent': '^0.3.1' },
  gitHead: 'c479846ccc1c244a14959fc93521ebf6f2f5f13e',
  scripts: {},
  dist: 
   { shasum: '61c0c6a72a35c3f47b011afe0d7620e4bb8c2ef3',
     tarball: 'http://registry.npmjs.org/smartystreets/-/smartystreets-1.2.0.tgz' },
  directories: {} }

~/smartystreets$ npm publish --force
npm WARN using --force I sure hope you know what you are doing.
npm WARN publish Forced publish over smartystreets@1.3.0
- smartystreets@1.3.0
npm ERR! Linux 3.16.0-31-generic
npm ERR! argv "node" "/redacted/usr/bin/npm" "publish" "--force"
npm ERR! node v0.10.25
npm ERR! npm  v2.5.1
npm ERR! code EPUBLISHCONFLICT

npm ERR! publish fail Cannot publish over existing version.
npm ERR! publish fail Update the 'version' field in package.json and try again.
npm ERR! publish fail 
npm ERR! publish fail To automatically increment version numbers, see:
npm ERR! publish fail     npm help version

npm ERR! Please include the following file with any support request:
npm ERR!     /redacted/smartystreets/npm-debug.log
~/smartystreets$ cat npm-debug.log
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/redacted/usr/bin/npm', 'publish', '--force' ]
2 info using npm@2.5.1
3 info using node@v0.10.25
4 verbose node symlink /usr/bin/node
5 warn using --force I sure hope you know what you are doing.
6 verbose publish [ '.' ]
7 silly cache add args [ '.', null ]
8 verbose cache add spec .
9 silly cache add parsed spec { raw: '.',
9 silly cache add   scope: null,
9 silly cache add   name: null,
9 silly cache add   rawSpec: '.',
9 silly cache add   spec: '/redacted/smartystreets',
9 silly cache add   type: 'directory' }
10 verbose addLocalDirectory /redacted/.npm/smartystreets/1.3.0/package.tgz not in flight; packing
11 verbose tar pack [ '/redacted/.npm/smartystreets/1.3.0/package.tgz',
11 verbose tar pack   '/redacted/smartystreets' ]
12 verbose tarball /redacted/.npm/smartystreets/1.3.0/package.tgz
13 verbose folder /redacted/smartystreets
14 info prepublish smartystreets@1.3.0
15 verbose addLocalTarball adding from inside cache /redacted/.npm/smartystreets/1.3.0/package.tgz
16 silly cache afterAdd smartystreets@1.3.0
17 verbose afterAdd /redacted/.npm/smartystreets/1.3.0/package/package.json not in flight; writing
18 verbose afterAdd /redacted/.npm/smartystreets/1.3.0/package/package.json written
19 silly publish { name: 'smartystreets',
19 silly publish   description: 'Fast SmartyStreets geocoder',
19 silly publish   version: '1.3.0',
19 silly publish   maintainers: [ { name: 'Tristan Davies', email: 'npm@tristan.io' } ],
19 silly publish   repository:
19 silly publish    { type: 'git',
19 silly publish      url: 'https://github.com/faradayio/node_smartystreets.git' },
19 silly publish   main: 'index.js',
19 silly publish   preferGlobal: 'true',
19 silly publish   bin: { smartystreets: 'program.js' },
19 silly publish   dependencies:
19 silly publish    { async: '^0.9.0',
19 silly publish      commander: '^2.7.1',
19 silly publish      'fast-csv': '^0.5.7',
19 silly publish      request: '^2.54.0',
19 silly publish      terminus: '^1.0.11',
19 silly publish      through2: '^0.6.3',
19 silly publish      'through2-concurrent': '^0.3.1' },
19 silly publish   readme: '# node_smartystreets\n\nA high-performance client for the SmartyStreets geocoding api\n\n## Dependencies\n\nnode and npm\n\n## Installation\n\nDepending on your environment you may need to run this as root in order to get the command line tool\n\n`npm install -g smartystreets`\n\n## Command line tool\n\nUsage: smartystreets [options]\n\n```\nOptions:\n\n  -h, --help                      output usage information\n  -V, --version                   output the version number\n  -i, --input [file]              Input (csv file) [stdin]\n  -o, --output [file]             Output (csv file) [stdout]\n  -s, --street-col [col]          Street col [street]\n  -z, --zipcode-col [col]         Zipcode col [zipcode]\n  -c, --city-col [col]            City col [city]\n  -S, --state-col [col]           State col [state]\n  --zipcode-filter [zips]         Only geocode records in certain zipcodes, comma separated\n  --state-filter [states]         Only geocode records in certain states, comma separated\n  -d, --delimiter [symbol]        CSV delimiter in input file\n  -O, --output-split [column]     Write to multiple files divided by column\n  --truncate-split [length]       Used with --output-split, truncate column to first X characters\n  -a, --auth-id [id]              SmartyStreets auth id [environment variable smartystreets_auth_id]\n  -A, --auth-token [token]        SmartyStreets auth token [environment variable smartystreets_auth_token]\n  -j, --concurrency [jobs]        Maximum number of concurrent requests [48]\n  -✈, --column-definition [mode]  Column definition mode or file [standard]\n  -p, --column-prefix [text]      Prefix for smartystreets columns in the output file [ss_]\n  -x, --column-suffix [text]      Suffix for smartystreets columns in the output file\n  -q, --quiet                     Quiet mode - turn off progress messages\n  -l, --log-interval [num]        Show progress after every X number of rows [1000]\n```\n\n### parameters\n\n`-i, --input [file]`\n\nNot required. Default stdin. If included it should be a path to a csv file.\n\n`-o, --output [file]`\n\nNot required. Default stdout. If included it should be a path to a csv file that does not exist yet.\n\n`-s, --street-col [col]`\n\nNot required. Default is \'street\'. It should be the name of the column in your input file that contains the first part of the address, e.g. \'123 main st\'\n\n`-z, --zipcode-col [col]`\n\nNot required. Default is \'zipcode\'. It should be the name of the column in your input file that contains the zipcode of an address, e.g. \'12345\'\n\n_Note_: if your file includes street and zipcode, you don\'t need to include city and state data, as it will not be used. If your file does not include zipcode, you _must_ include city _and_ state columns.\n\n`-c, --city-col [col]`\n\nNot required. Default is \'city\'. It should be the name of the column in your input file that contains the city or town, e.g. \'Albany\'\n\n_Note_: city is only used if zipcode is not present\n\n`-S, --state-col [col]`\n\nNot required. Default is \'state\'. It should be the name of the column in your input file that contains the state, e.g. \'Vermont\'\n\n`--zipcode-filter [zips]`\n\nNot required. Only geocodes records in certain zipcodes. Separate by comma for multiple zips. Example: `--zipcode-filter 05401,05403`. Uses column specified in `--zipcode-col`.\n\n`--state-filter [zips]`\n\nNot required. Only geocodes records in certain states. Separate by comma for multiple states. Example: `--state-filter VT,NY`. Uses column specified in `--state-col`.\n\n`-O, --output-split [column]`\n\nNot required. This will write multiple output files, split by a column value. `--output-split ss_zipcode` will write one file per zipcode, under `output/12345.csv`.\n\n`--truncate-split [length]`\n\nNot required. Used with `--output-split`. It will truncate the column value to a number of characters. `--output-split ss_zipcode --truncate-split 3` will write files like `output/123.csv` which will contain all rows with zipcodes that start with 123.\n\n`-a, --auth-id [id]`\n\n_Required_. You can get this [here](https://smartystreets.com/account/keys) listed under "Security Keys"\n\n`-A, --auth-token [token]`\n\n_Required_. You can get this [here](https://smartystreets.com/account/keys) listed under "Security Keys"\n\n`-j, --concurrency [jobs]`\n\nNot required. Default 48. Tune this if you fancy yourself an engineer and want to get a slight increase in performance. It should be higher if you have an unmetered 10 gigabit connection, and lower if you\'re on an EDGE wireless signal. If you want to run fifty instances of this program at the same time, you may experience dropped connections which can be dealt with by turning this down.\n\n`-✈, --column-definition [mode|file|string]`\n\nNot required. Default standard. This defines which columns from smartystreets will be added to your output file. You can find the definitions in the `structure` directory. `mail` will get you a mailing address and nothing more, and `complete` will get you every column which smartystreets has to offer. You can also make your own custom definition based on one of the existing definitions and use that by passing in the path. Also, if you\'re insane, you can just pass it JSON directly. I don\'t even know why I added that option. Probably had a good reason at the time. Don\'t even know if it works. I should add tests.\n\n`-p, --column-prefix [text]`\n\nNot required. Default ss_. This is the prefix applied to all columns that are added to your output file. `delivery_line_1` becomes `ss_delivery_line_1`\n\n`-x, --column-suffix [text]`\n\nNot required. Default empty string. This is the suffix applied to all columns that are added to your output file. `delivery_line_1` could become `delivery_line_1_suffixGoesHere`\n\n## Geocoding stream API\n\nThis module also provides a simple api that you can use in your own node projects.\n\n```\nvar Smartystreets = require(\'smartystreets\');\nvar fs = require(\'fs\');\n\nvar geocoder = new Smartystreets(options);\n\nfs.createReadStream(\'input.csv\')\n  .pipe(geocoder)\n  .pipe(process.stdout);\n```\n\nIn this example, `geocoder` is a duplex stream which will take every address that is written to it in CSV format and output a geocoded CSV.\n\nThe `options` argument is identical to the options used by the command line tool, except that camelCase is used (street-col -> streetCol) and "input" and "output" are ignored (instead use the piping solution above).\n\n## Notes\n\n* This is designed for performance and if you run it on the wrong file you might use up your entire api quota in a matter of seconds. If you have a limited number of lookups, be careful.\n* Since many requests are made in parallel, it\'s unlikely that the rows of the output CSV will be in the same order as the input CSV.\n\n## Authors\n\n* [Tristan Davies](mailto:npm@tristan.io)\n* [Seamus Abshere](mailto:seamus@abshere.net)\n\n## Corporate support\n\n<p><a href="http://faraday.io" alt="Faraday"><img src="https://s3.amazonaws.com/creative.faraday.io/logo.png" alt="Faraday logo"/></a></p>\n\n## Copyright\n\nCopyright 2015 [Faraday](http://faraday.io)\n',
19 silly publish   readmeFilename: 'README.md',
19 silly publish   gitHead: '5abca7b9848c4546e4233e93a44d967da6601ecb',
19 silly publish   bugs: { url: 'https://github.com/faradayio/node_smartystreets/issues' },
19 silly publish   homepage: 'https://github.com/faradayio/node_smartystreets',
19 silly publish   _id: 'smartystreets@1.3.0',
19 silly publish   scripts: {},
19 silly publish   _shasum: '734928782827cd26efb15a092874fa176dcd9445',
19 silly publish   _from: '.' }
20 silly mapToRegistry name smartystreets
21 silly mapToRegistry using default registry
22 silly mapToRegistry registry https://registry.npmjs.org/
23 silly mapToRegistry uri https://registry.npmjs.org/smartystreets
24 verbose publish registryBase https://registry.npmjs.org/
25 silly publish uploading /redacted/.npm/smartystreets/1.3.0/package.tgz
26 verbose request uri https://registry.npmjs.org/smartystreets
27 verbose request sending authorization for write operation
28 info attempt registry request try #1 at 09:45:22
29 verbose request id 318c43e010c5bfbb
30 http request PUT https://registry.npmjs.org/smartystreets
31 http 409 https://registry.npmjs.org/smartystreets
32 verbose headers { date: 'Fri, 27 Mar 2015 13:45:22 GMT',
32 verbose headers   server: 'Apache',
32 verbose headers   'content-type': 'application/json',
32 verbose headers   'cache-control': 'max-age=60',
32 verbose headers   'content-length': '17',
32 verbose headers   'accept-ranges': 'bytes',
32 verbose headers   via: '1.1 varnish',
32 verbose headers   'x-served-by': 'cache-iad2121-IAD',
32 verbose headers   'x-cache': 'MISS',
32 verbose headers   'x-cache-hits': '0',
32 verbose headers   'x-timer': 'S1427463922.523930,VS0,VE379',
32 verbose headers   'keep-alive': 'timeout=10, max=50',
32 verbose headers   connection: 'Keep-Alive' }
33 verbose request invalidating /redacted/.npm/registry.npmjs.org/smartystreets on PUT
34 verbose request uri https://registry.npmjs.org/smartystreets?write=true
35 verbose request no auth needed
36 info attempt registry request try #1 at 09:45:23
37 http request GET https://registry.npmjs.org/smartystreets?write=true
38 http 200 https://registry.npmjs.org/smartystreets?write=true
39 warn publish Forced publish over smartystreets@1.3.0
40 silly unpublish args[0] smartystreets@1.3.0
41 silly unpublish thing { raw: 'smartystreets@1.3.0',
41 silly unpublish   scope: null,
41 silly unpublish   name: 'smartystreets',
41 silly unpublish   rawSpec: '1.3.0',
41 silly unpublish   spec: '1.3.0',
41 silly unpublish   type: 'version' }
42 silly ls normalized smartystreets/1.3.0
43 verbose gentlyRm vacuuming /redacted/.npm/smartystreets/1.3.0
44 silly gentlyRm purging /redacted/.npm/smartystreets/1.3.0
45 silly gentlyRm quitting because other entries in /redacted/.npm/smartystreets
46 silly mapToRegistry name smartystreets
47 silly mapToRegistry using default registry
48 silly mapToRegistry registry https://registry.npmjs.org/
49 silly mapToRegistry uri https://registry.npmjs.org/smartystreets
50 verbose get GET as part of write; not caching result
51 verbose request uri https://registry.npmjs.org/smartystreets?write=true
52 verbose request no auth needed
53 info attempt registry request try #1 at 09:45:23
54 http request GET https://registry.npmjs.org/smartystreets?write=true
55 http 200 https://registry.npmjs.org/smartystreets?write=true
56 silly get cb [ 200,
56 silly get   { date: 'Fri, 27 Mar 2015 13:45:23 GMT',
56 silly get     server: 'CouchDB/1.5.0 (Erlang OTP/R16B03)',
56 silly get     etag: '"2P92VYX14CWVEJSMQORH6M59S"',
56 silly get     'content-type': 'application/json',
56 silly get     'cache-control': 'max-age=0',
56 silly get     'content-length': '49194',
56 silly get     'accept-ranges': 'bytes',
56 silly get     via: '1.1 varnish',
56 silly get     age: '0',
56 silly get     'x-served-by': 'cache-iad2141-IAD',
56 silly get     'x-cache': 'MISS',
56 silly get     'x-cache-hits': '0',
56 silly get     'x-timer': 'S1427463923.420276,VS0,VE30',
56 silly get     vary: 'Accept',
56 silly get     'keep-alive': 'timeout=10, max=50',
56 silly get     connection: 'Keep-Alive' } ]
57 verbose get saving smartystreets to /redacted/.npm/registry.npmjs.org/smartystreets_3Fwrite_3Dtrue/.cache.json
58 info unpublish https://registry.npmjs.org/smartystreets@1.3.0 not published
59 verbose publish [ '.' ]
60 silly cache add args [ '.', null ]
61 verbose cache add spec .
62 silly cache add parsed spec { raw: '.',
62 silly cache add   scope: null,
62 silly cache add   name: null,
62 silly cache add   rawSpec: '.',
62 silly cache add   spec: '/redacted/smartystreets',
62 silly cache add   type: 'directory' }
63 verbose addLocalDirectory /redacted/.npm/smartystreets/1.3.0/package.tgz not in flight; packing
64 verbose tar pack [ '/redacted/.npm/smartystreets/1.3.0/package.tgz',
64 verbose tar pack   '/redacted/smartystreets' ]
65 verbose tarball /redacted/.npm/smartystreets/1.3.0/package.tgz
66 verbose folder /redacted/smartystreets
67 info prepublish smartystreets@1.3.0
68 verbose addLocalTarball adding from inside cache /redacted/.npm/smartystreets/1.3.0/package.tgz
69 silly cache afterAdd smartystreets@1.3.0
70 verbose afterAdd /redacted/.npm/smartystreets/1.3.0/package/package.json not in flight; writing
71 verbose afterAdd /redacted/.npm/smartystreets/1.3.0/package/package.json written
72 silly publish { name: 'smartystreets',
72 silly publish   description: 'Fast SmartyStreets geocoder',
72 silly publish   version: '1.3.0',
72 silly publish   maintainers: [ { name: 'tristandavies', email: 'npm@tristan.io' } ],
72 silly publish   repository:
72 silly publish    { type: 'git',
72 silly publish      url: 'https://github.com/faradayio/node_smartystreets.git' },
72 silly publish   main: 'index.js',
72 silly publish   preferGlobal: 'true',
72 silly publish   bin: { smartystreets: 'program.js' },
72 silly publish   dependencies:
72 silly publish    { async: '^0.9.0',
72 silly publish      commander: '^2.7.1',
72 silly publish      'fast-csv': '^0.5.7',
72 silly publish      request: '^2.54.0',
72 silly publish      terminus: '^1.0.11',
72 silly publish      through2: '^0.6.3',
72 silly publish      'through2-concurrent': '^0.3.1' },
72 silly publish   readme: '# node_smartystreets\n\nA high-performance client for the SmartyStreets geocoding api\n\n## Dependencies\n\nnode and npm\n\n## Installation\n\nDepending on your environment you may need to run this as root in order to get the command line tool\n\n`npm install -g smartystreets`\n\n## Command line tool\n\nUsage: smartystreets [options]\n\n```\nOptions:\n\n  -h, --help                      output usage information\n  -V, --version                   output the version number\n  -i, --input [file]              Input (csv file) [stdin]\n  -o, --output [file]             Output (csv file) [stdout]\n  -s, --street-col [col]          Street col [street]\n  -z, --zipcode-col [col]         Zipcode col [zipcode]\n  -c, --city-col [col]            City col [city]\n  -S, --state-col [col]           State col [state]\n  --zipcode-filter [zips]         Only geocode records in certain zipcodes, comma separated\n  --state-filter [states]         Only geocode records in certain states, comma separated\n  -d, --delimiter [symbol]        CSV delimiter in input file\n  -O, --output-split [column]     Write to multiple files divided by column\n  --truncate-split [length]       Used with --output-split, truncate column to first X characters\n  -a, --auth-id [id]              SmartyStreets auth id [environment variable smartystreets_auth_id]\n  -A, --auth-token [token]        SmartyStreets auth token [environment variable smartystreets_auth_token]\n  -j, --concurrency [jobs]        Maximum number of concurrent requests [48]\n  -✈, --column-definition [mode]  Column definition mode or file [standard]\n  -p, --column-prefix [text]      Prefix for smartystreets columns in the output file [ss_]\n  -x, --column-suffix [text]      Suffix for smartystreets columns in the output file\n  -q, --quiet                     Quiet mode - turn off progress messages\n  -l, --log-interval [num]        Show progress after every X number of rows [1000]\n```\n\n### parameters\n\n`-i, --input [file]`\n\nNot required. Default stdin. If included it should be a path to a csv file.\n\n`-o, --output [file]`\n\nNot required. Default stdout. If included it should be a path to a csv file that does not exist yet.\n\n`-s, --street-col [col]`\n\nNot required. Default is \'street\'. It should be the name of the column in your input file that contains the first part of the address, e.g. \'123 main st\'\n\n`-z, --zipcode-col [col]`\n\nNot required. Default is \'zipcode\'. It should be the name of the column in your input file that contains the zipcode of an address, e.g. \'12345\'\n\n_Note_: if your file includes street and zipcode, you don\'t need to include city and state data, as it will not be used. If your file does not include zipcode, you _must_ include city _and_ state columns.\n\n`-c, --city-col [col]`\n\nNot required. Default is \'city\'. It should be the name of the column in your input file that contains the city or town, e.g. \'Albany\'\n\n_Note_: city is only used if zipcode is not present\n\n`-S, --state-col [col]`\n\nNot required. Default is \'state\'. It should be the name of the column in your input file that contains the state, e.g. \'Vermont\'\n\n`--zipcode-filter [zips]`\n\nNot required. Only geocodes records in certain zipcodes. Separate by comma for multiple zips. Example: `--zipcode-filter 05401,05403`. Uses column specified in `--zipcode-col`.\n\n`--state-filter [zips]`\n\nNot required. Only geocodes records in certain states. Separate by comma for multiple states. Example: `--state-filter VT,NY`. Uses column specified in `--state-col`.\n\n`-O, --output-split [column]`\n\nNot required. This will write multiple output files, split by a column value. `--output-split ss_zipcode` will write one file per zipcode, under `output/12345.csv`.\n\n`--truncate-split [length]`\n\nNot required. Used with `--output-split`. It will truncate the column value to a number of characters. `--output-split ss_zipcode --truncate-split 3` will write files like `output/123.csv` which will contain all rows with zipcodes that start with 123.\n\n`-a, --auth-id [id]`\n\n_Required_. You can get this [here](https://smartystreets.com/account/keys) listed under "Security Keys"\n\n`-A, --auth-token [token]`\n\n_Required_. You can get this [here](https://smartystreets.com/account/keys) listed under "Security Keys"\n\n`-j, --concurrency [jobs]`\n\nNot required. Default 48. Tune this if you fancy yourself an engineer and want to get a slight increase in performance. It should be higher if you have an unmetered 10 gigabit connection, and lower if you\'re on an EDGE wireless signal. If you want to run fifty instances of this program at the same time, you may experience dropped connections which can be dealt with by turning this down.\n\n`-✈, --column-definition [mode|file|string]`\n\nNot required. Default standard. This defines which columns from smartystreets will be added to your output file. You can find the definitions in the `structure` directory. `mail` will get you a mailing address and nothing more, and `complete` will get you every column which smartystreets has to offer. You can also make your own custom definition based on one of the existing definitions and use that by passing in the path. Also, if you\'re insane, you can just pass it JSON directly. I don\'t even know why I added that option. Probably had a good reason at the time. Don\'t even know if it works. I should add tests.\n\n`-p, --column-prefix [text]`\n\nNot required. Default ss_. This is the prefix applied to all columns that are added to your output file. `delivery_line_1` becomes `ss_delivery_line_1`\n\n`-x, --column-suffix [text]`\n\nNot required. Default empty string. This is the suffix applied to all columns that are added to your output file. `delivery_line_1` could become `delivery_line_1_suffixGoesHere`\n\n## Geocoding stream API\n\nThis module also provides a simple api that you can use in your own node projects.\n\n```\nvar Smartystreets = require(\'smartystreets\');\nvar fs = require(\'fs\');\n\nvar geocoder = new Smartystreets(options);\n\nfs.createReadStream(\'input.csv\')\n  .pipe(geocoder)\n  .pipe(process.stdout);\n```\n\nIn this example, `geocoder` is a duplex stream which will take every address that is written to it in CSV format and output a geocoded CSV.\n\nThe `options` argument is identical to the options used by the command line tool, except that camelCase is used (street-col -> streetCol) and "input" and "output" are ignored (instead use the piping solution above).\n\n## Notes\n\n* This is designed for performance and if you run it on the wrong file you might use up your entire api quota in a matter of seconds. If you have a limited number of lookups, be careful.\n* Since many requests are made in parallel, it\'s unlikely that the rows of the output CSV will be in the same order as the input CSV.\n\n## Authors\n\n* [Tristan Davies](mailto:npm@tristan.io)\n* [Seamus Abshere](mailto:seamus@abshere.net)\n\n## Corporate support\n\n<p><a href="http://faraday.io" alt="Faraday"><img src="https://s3.amazonaws.com/creative.faraday.io/logo.png" alt="Faraday logo"/></a></p>\n\n## Copyright\n\nCopyright 2015 [Faraday](http://faraday.io)\n',
72 silly publish   readmeFilename: 'README.md',
72 silly publish   gitHead: '5abca7b9848c4546e4233e93a44d967da6601ecb',
72 silly publish   bugs: { url: 'https://github.com/faradayio/node_smartystreets/issues' },
72 silly publish   homepage: 'https://github.com/faradayio/node_smartystreets',
72 silly publish   _id: 'smartystreets@1.3.0',
72 silly publish   scripts: {},
72 silly publish   _shasum: '734928782827cd26efb15a092874fa176dcd9445',
72 silly publish   _from: '.',
72 silly publish   _npmVersion: '2.5.1',
72 silly publish   _nodeVersion: '0.10.25',
72 silly publish   _npmUser: { name: 'tristandavies', email: 'npm@tristan.io' },
72 silly publish   dist:
72 silly publish    { shasum: '734928782827cd26efb15a092874fa176dcd9445',
72 silly publish      tarball: 'http://registry.npmjs.org/smartystreets/-/smartystreets-1.3.0.tgz' } }
73 silly mapToRegistry name smartystreets
74 silly mapToRegistry using default registry
75 silly mapToRegistry registry https://registry.npmjs.org/
76 silly mapToRegistry uri https://registry.npmjs.org/smartystreets
77 verbose publish registryBase https://registry.npmjs.org/
78 silly publish uploading /redacted/.npm/smartystreets/1.3.0/package.tgz
79 verbose request uri https://registry.npmjs.org/smartystreets
80 verbose request sending authorization for write operation
81 info attempt registry request try #1 at 09:45:24
82 http request PUT https://registry.npmjs.org/smartystreets
83 http 409 https://registry.npmjs.org/smartystreets
84 verbose headers { date: 'Fri, 27 Mar 2015 13:45:24 GMT',
84 verbose headers   server: 'Apache',
84 verbose headers   'content-type': 'application/json',
84 verbose headers   'cache-control': 'max-age=60',
84 verbose headers   'content-length': '17',
84 verbose headers   'accept-ranges': 'bytes',
84 verbose headers   via: '1.1 varnish',
84 verbose headers   'x-served-by': 'cache-iad2141-IAD',
84 verbose headers   'x-cache': 'MISS',
84 verbose headers   'x-cache-hits': '0',
84 verbose headers   'x-timer': 'S1427463923.760607,VS0,VE1091',
84 verbose headers   'keep-alive': 'timeout=10, max=50',
84 verbose headers   connection: 'Keep-Alive' }
85 verbose request invalidating /redacted/.npm/registry.npmjs.org/smartystreets on PUT
86 verbose request uri https://registry.npmjs.org/smartystreets?write=true
87 verbose request no auth needed
88 info attempt registry request try #1 at 09:45:25
89 http request GET https://registry.npmjs.org/smartystreets?write=true
90 http 200 https://registry.npmjs.org/smartystreets?write=true
91 verbose stack Error: cannot modify pre-existing version
91 verbose stack     at conflictError (/redacted/usr/lib/node_modules/npm/node_modules/npm-registry-client/lib/publish.js:184:11)
91 verbose stack     at CachingRegistryClient.putNext (/redacted/usr/lib/node_modules/npm/node_modules/npm-registry-client/lib/publish.js:147:15)
91 verbose stack     at /redacted/usr/lib/node_modules/npm/node_modules/npm-registry-client/lib/publish.js:129:15
91 verbose stack     at /redacted/usr/lib/node_modules/npm/lib/cache/caching-client.js:52:8
91 verbose stack     at f (/redacted/usr/lib/node_modules/npm/node_modules/once/once.js:17:25)
91 verbose stack     at /redacted/usr/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:73:16
91 verbose stack     at f (/redacted/usr/lib/node_modules/npm/node_modules/once/once.js:17:25)
91 verbose stack     at CachingRegistryClient.<anonymous> (/redacted/usr/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:262:12)
91 verbose stack     at Request._callback (/redacted/usr/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:178:14)
91 verbose stack     at Request.self.callback (/redacted/usr/lib/node_modules/npm/node_modules/request/request.js:344:22)
92 verbose pkgid smartystreets
93 verbose cwd /redacted/smartystreets
94 error Linux 3.16.0-31-generic
95 error argv "node" "/redacted/usr/bin/npm" "publish" "--force"
96 error node v0.10.25
97 error npm  v2.5.1
98 error code EPUBLISHCONFLICT
99 error publish fail Cannot publish over existing version.
99 error publish fail Update the 'version' field in package.json and try again.
99 error publish fail
99 error publish fail To automatically increment version numbers, see:
99 error publish fail     npm help version
100 verbose exit [ 1, true ]
```

**Comment by bcoe** [comment]
@devTristan did you get an error when you initially published `1.3.0`., or was this your first time attempting to publish it?

**Comment by bcoe** [comment]
@devTristan one way a package can get in this state is if you unpublish a version. Even if you unpublish the prior version, it's necessary to bump the version of the next package you publish.

**Comment by devTristan** [comment]
I don't believe I unpublished it prior to this publish. I attempted to publish it yesterday and got the same error, but did not document the issue until today.

I did attempt to unpublish it after posting this issue, but ended up bumping the version.

**Comment by bcoe** [comment]
@devTristan I see 1.3.2 is successfully published., so version bumping did the trick for you?

I'm going to close this issue for the time being., but please let me know if this happens again. I'd like to see JSON prior to you performing an unpublish.

**Comment by devTristan** [comment]
Version bumping did the trick. Chances are I messed up the original publish somehow, or unpublished it and forgot. I'll create a new issue if I see this issue again.
